### PR TITLE
Get rid of unnecessary grouping operators

### LINKED ISSUE #5: Unnecessary parentheses
Repost from TimvanScherpenzeel/glsl-minifier#2.

### PR DESCRIPTION
**Pull request by axion014**
Closes #5.

### CODE REVIEW

**Comment by dhritzkiv** [comment]
Hey @axion014. Thanks for this and your other commits.

I just updated the repo to include tests. Do you suppose you could add a test in `test/basic.js`? 

Feel free to add simpler .glsl files to the test directory that helps you test your new code. Also add some assertions that validate that no edge-cases or gotchas arise.

File: index.js [hunk 1]
```diff
@@ -23,6 +23,16 @@ function minifier(safe_words, mutate_storages) {
   return through(mutate)
 
   function mutate(node) {
+    // remove unnecessary grouping operators
+    if(is_unnecessary_group(node)) return
+
+    if(node.parent) for(var current = node.parent; current.parent; current = current.parent) {
```
Comment (dhritzkiv) [inline index.js]: For stylistic/readability purposes, can you give this `if` statement braces, and encapsulate the `for` loop within it?

**Comment by axion014** [comment]
I just found it tricky to get right. For a minifier I would want it to recognize when a operator is commutative.
However, the same `*` operator can be used both on scalar and matrix, which makes it difficult.

**Comment by dhritzkiv** [comment]
Feel free to implement tests for #7 and #8 before coming back to this one

**Comment by dhritzkiv** [comment]
I updated the tests here as well to be more readable + succinct

**Comment by dhritzkiv** [comment]
Btw, does `-` qualify as a commutative operator?

**Comment by axion014** [comment]
> Btw, does `-` qualify as a commutative operator?

It doesn't - `1 - 2 !== 2 - 1`

**Comment by dhritzkiv** [comment]
Duh!

**Comment by axion014** [comment]
Thanks for your time on maintaining this repository. You appear to be going even further, eh?

**Comment by dhritzkiv** [comment]
@axion014 No problem. I'll push a release soon.

Your redundant vec parameters feature inspired me to apply it to mats! Mind taking a look and seeing if it makes sense and if it would be useful to you? #9
